### PR TITLE
Use run instead of popen to run ar [blocks: #3867]

### PR DESCRIPTION
**Pull request by tautschnig**
Loading files from archives requires the use of ar; invoke it via run instead of
popen to eliminate OS-specific code from the goto-cc folder.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/goto-gcc/CMakeLists.txt [hunk 1]
```diff
@@ -2,3 +2,14 @@
 # isn't marked as an executable (target), which CMake requires. Thus construct a
 # path in the same way the symbolic link is created in the goto-cc directory.
 add_test_pl_tests("$<TARGET_FILE_DIR:goto-cc>/goto-gcc")
+
+add_custom_command(OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/archives/libour_archive.a"
+    COMMAND "$<TARGET_FILE_DIR:goto-cc>/goto-gcc" -c foo.c
+    COMMAND ${CMAKE_AR} rcs libour_archive.a foo.o
+    DEPENDS "$<TARGET_FILE_DIR:goto-cc>/goto-gcc"
```
Comment (smowton) [inline regression/goto-gcc/CMakeLists.txt]: Also depends on foo.c

**Comment by smowton** [comment]
(the rest of this is hard for me to review in a timely fashion without reading up a bunch on OSX object formats and such)

File: regression/goto-gcc/CMakeLists.txt (hunk 1 above)
Comment (tautschnig) [reply regression/goto-gcc/CMakeLists.txt]: Ah, thanks, added.

**Comment by tautschnig** [comment]
This PR is now restricted to what it said in the subject. Mach-O binary support, which is needed to properly test it in CI, will be done in a separate PR.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1b947fe0798c72b5c1add38dae28dbe9062123cd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/98025595

**Review by smowton — APPROVED** [review]

**Review by kroening — APPROVED** [review]

**Pull request merged**